config/routes.rb

```diff
@@ -21,7 +21,8 @@
   end
 
   get ':username' => 'users/lists#index', as: :user_profile
-  get ':username/:list_slug' => 'users/lists#show', as: :user_list
-  get ':username/:list_slug/edit' => 'lists#edit', as: :edit_user_list
-  get ':username/:list_slug/:reference_id' => 'users/references#show', as: :user_reference
+  get ':username/:id' => 'users/lists#show', as: :user_list
+  get ':username/:id/edit' => 'lists#edit', as: :edit_user_list
+  get ':username/:id/destroy' => 'lists#destroy', as: :destroy_user_list
+  get ':username/:id/:reference_id' => 'users/references#show', as: :user_reference
```

Comment:
this one should be `:list_id` and `:reference_id` should be `:id` according to rails conventions